srcpkgs/qt6-base/template

```diff
@@ -0,0 +1,194 @@
+# Template file for 'qt6-base'
+pkgname=qt6-base
+version=6.1.0rc2
+revision=1
+wrksrc="qtbase-everywhere-src-${version/rc/-rc}"
+build_style=cmake
+configure_args="-DINSTALL_DATADIR=share/qt6
+ -DINSTALL_ARCHDATADIR=lib/qt6
+ -DINSTALL_BINDIR=lib/qt6/bin
+ -DINSTALL_MKSPECSDIR=lib/qt6/mkspecs
+ -DINSTALL_INCLUDEDIR=include/qt6
+ -DINSTALL_SYSCONFDIR=/etc/xdg
+ -DQT_USE_CCACHE=ON"
+hostmakedepends="perl pkg-config xmlstarlet"
+makedepends="zlib-devel libzstd-devel dbus-devel
+ double-conversion-devel glib-devel libxcb-devel
+ xcb-util-image-devel xcb-util-keysyms-devel
+ xcb-util-renderutil-devel xcb-util-wm-devel
+ MesaLib-devel pcre2-devel icu-devel mtdev-devel harfbuzz-devel
+ libpng-devel tslib-devel libinput-devel gtk+3-devel
+ libmariadbclient-devel postgresql-libs-devel
+ unixodbc-devel cups-devel libproxy-devel brotli-devel
+ sqlite-devel Vulkan-Headers mit-krb5-devel vulkan-loader"
+short_desc="Cross-platform application and UI framework (QT6)"
+maintainer="John <me@johnnynator.dev>"
+license="GPL-3.0-only with Qt-GPL-exception-1.0, LGPL-3.0-only, GPL-2.0-or-later"
+homepage="https://www.qt.io"
+distfiles="https://download.qt.io/development_releases/qt/${version%.*}/${version/rc/-rc}/submodules/qtbase-everywhere-src-${version/rc/-rc}.tar.xz"
+checksum=3884c64750a9b6ef544ee2fb72a2bf5e5e167501d6ea11ab1a55105e354a506e
+python_version=3
+
+# TODO: fix usage of ccache
```

Comment:
I had a look on qt6-base and cmake. here my points:

"`-D-DQT_USE_CCACHE`" has no effect if ccache is enabled from xbps-src  : ccache will be used.
I think "`-DQT_USE_CCACHE=OFF`" might be a better idea (similar to other packages)

qt6-base build uses precompiled headers (pch) when working with automoc. ccache does not handle pch until some additional configuration (see https://ccache.dev/manual/4.2.1.html#_precompiled_headers).
 => so ccache is rather useless in this case: about 150/1160 "objects" are stored in ccache
workarounds:
* hope that packagers will enable pch support in ccache. IMHO, this support is somewhat imprecise, afraid of false positive
* disable use of pch for building qt6-base with "`-DBUILD_WITH_PCH=OFF`" : some overhead (not that much) on first build, but ccache efficiency after (about 70 minutes on 1st run, 6 minutes  then to build+package qt6-base). And no particular settings